lib/bolt/inventory/group.rb

```diff
@@ -315,7 +316,26 @@ def resolve_data_keys(data, target = nil)
           'features' => @plugins.resolve_references(data.fetch('features', [])),
           'plugin_hooks' => @plugins.resolve_references(data.fetch('plugin_hooks', {}))
         }
+
         validate_data_keys(result, target)
+
+        Bolt::Config::Options::TRANSPORT_CONFIG.each_key do |transport|
+          next unless result['config'].key?(transport)
+          transport_config = result['config'][transport]
+          next unless transport_config.is_a?(Hash)
+          transport_config = Bolt::Util.postwalk_vals(transport_config) do |val|
+            if val.is_a?(Hash)
+              val.compact!
```

Comment:
I'm concerned about making in-place updates to an object that we didn't necessarily construct. Specifically I'm unsure how this will affect future targets getting their config from the same group. We should make a copy here I think.

Reply:
I also think we don't need to prune empty hashes, since we do a deep merge. Empty hashes won't have any affect on the final merged result in that case. Combined with the previous comment, that would make this block just

```rb
if val.is_a?(Hash)
  val.compact
else
  val
end
```

Potentially we would prefer not to make a copy of the hash in the common case where it _doesn't_ contain any `nil` values, but I don't think that matters too much.

Reply:
I'm fine with changing to a copy.

Re: the second comment, would returning an empty hash still be OK for _target_ config? E.g. will a target-config that's something like `{"ssh": {"private-key": {}}}` cause any issues?

Reply:
Yeah i think the private-key case is somewhat special. currently if you allow it to be an empty hash you get a schema error. That is a weird one but i imagine it would be a common one for connect.

Reply:
In that case, I lean towards the current behavior of deleting keys that correspond to empty hashes.